### ISSUE TITLE
Segmentation Fault: loss.backward() for mobile net_v2 trained on CIFAR100

### ISSUE THREAD

**Issue by datafool** [pytorch/pytorch#22592]
## 🐛 Bug

I am trying to train pretrained mobilenet_v2() on CIFAR 100 dataset as taken from torch vision. However, loss.backward() fails and crashes python with `Segmentation Fault 11` error. Same error is not observed when using any other pretrained model like VGG16 etc.
## To Reproduce

Please use below code snippet to reproduce the error

```
class MobileNet(nn.Module):
    def __init__(self):
        super(MobileNet, self).__init__()
        
        self.mobile = models.mobilenet_v2(pretrained=True)
        in_features = self.mobile.classifier[-1].in_features
        out_features = 100
        self.mobile.classifier[-1] = nn.Linear(in_features=in_features, out_features=out_features)
        self.softmax = nn.Softmax(dim=1)
        
    def forward(self, x):
        x = self.mobile(x)
        return self.softmax(x)

train_transform = transforms.Compose([
    transforms.Resize(256),
    transforms.CenterCrop(224),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406], 
                         std=[0.229, 0.224, 0.225]),
])

train_dataset = torchvision.datasets.CIFAR100(root=path, 
                                              train=True, 
                                              download=True, 
                                              transform=train_transform)

train_loader = torch.utils.data.DataLoader(train_dataset, 
                                           shuffle=True, 
                                           batch_size=4)

model = MobileNet()
optimzer = torch.optim.Adam(model.parameters())
criterion = nn.CrossEntropyLoss(reduction='sum')

i, (image, labels) = next(enumerate(train_loader))

out = model(image)

loss = criterion(out, labels)
loss.backward()
```

## Expected behavior

Model should back-propogate without segmentation fault.

## Environment

PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.4
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy==1.16.4
[pip] torch==1.1.0
[pip] torchvision==0.3.0
[conda] mkl                       2019.4                      233
[conda] pytorch                   1.1.0                   py3.7_0    pytorch
[conda] torchvision               0.3.0             py37_cuNone_1    pytorch

**Comment by NadavLightricks** [comment]
I'm experiencing the same issue (only I use mobilenet_v2 for imagenet).
I think the problem is with the convolution layers that implement depthwise separable convolution. Specifically, in the InvertedResidual class, if I change `groups=hidden_dim` to something like `groups=hidden_dim // 2`, training goes smoothly.

It seems that the case where "groups" is as small as possible messes something up.

**Comment by soumith** [comment]
Between this and https://github.com/pytorch/pytorch/issues/23487 enough sets of users have reported this. A segmentation fault needs to be fixed immediately, and atleast has to be brought to a reasonable error message.

**Comment by soumith** [comment]
a fix is part of master, and part of nightlies.
this will be parts of the upcoming release next week.
